### ISSUE TITLE
Overlay displayed even when disabled

### ISSUE THREAD

**Issue by pauloup** [tribalfs/GalaxyMaxHzPub#51]
**Describe the bug**
Once the overlay is displayed by enabling the option "Refresh rate monitor" and selecting "Overlay" or "Both", it will stay on the screen, even after selecting "Statusbar" or disabling "Refresh rate monitor".

![Screenshot_2022-09-20-10-13-47-580-edit_com google android apps photos](https://user-images.githubusercontent.com/17508852/191267260-94bf6d75-56c5-4b87-92f6-544548de7f99.jpg)

**To Reproduce**

A. GMH version and the exact settings in GMH
v8.0.0. The issue seems to happen with any setting.

B. Steps to reproduce the behavior:
1. Enable "Refresh rate monitor".
2. Select "Overlay" or "Both".
3. The overlay shows.
4. Select "Statusbar".
5. The overlay still shows.
6. Disable "Refresh rate monitor".
7. The overlay still shows.

**Workaround**

The only way I've found to hide the overlay:
1. Enable "Refresh rate monitor".
2. Force close the app.
3. The overlay hides.
4. Open the app.
5. Enable "Refresh rate monitor".
6. The overlay stays hidden.

**Expected behavior**
The overlay should be hidden from screen when "Statusbar" is selected, or "Refresh rate monitor" is disabled.

**Screen Recording**

https://photos.app.goo.gl/M5LCDDHVSzq7bF4A8

**Smartphone**
 - Model Number: POCO F4 GT
 - OS: Android 12, xiaomi.eu 13.0.11 (MIUI)
 - Date of latest security update: August 2022

Note: I've read the FAQ and all the past issues, but couldn't find anything. Sorry if it was addressed before.

**Comment by tribalfs** [comment]
If you are coming from v7, clear data is required.

**Comment by pauloup** [comment]
I'm not, it was my first time installing Galaxy Max Hz

**Comment by tribalfs** [comment]
You are using a non-samsung device and this might be the cause.
But try clearing app data.

**Comment by pauloup** [comment]
Before starting the issue, and while looking for a workaround, I've tried both restarting the phone, and doing a clear data.

Unfortunately, none seemed to help. As soon the app started, the overlay was back on screen.

Then I've uninstalled and reinstalled the app. That seemed to help, and the overlay didn't returned.

But as soon as I tried once more enabling the Overlay option, I couldn't hidden it afterwards. The only workaround I've found was to disable "Refresh rate monitor" and force close the app.

---

That's true that I'm not on a Samsung device, and I understand you may not be able to help me in this case. 

But I still would like to thank you for this app. It worked perfectly on my phone, much better than the native Adaptive mode on the system, that keeps my screen on 120hz almost all the time. I even wrote a [Tutorial](https://medium.com/@paulotwain/tutorial-galaxy-max-hz-e6aab58265a6) to recommend the app to other Poco F4 GT users, and shared it on the Brazilian Telegram group of the phone.

Finally, I've looked at the logcat, and found many of these
![Screenshot_2022-09-20-11-09-52-741_com dp logcatapp](https://user-images.githubusercontent.com/17508852/191280384-12aab137-c8c1-43b1-acf7-d50b70338642.png)
So I think you're right, it seems to be a MIUI issue.

**Comment by tribalfs** [comment]
Glad to know that.
Thank you.

If we have spare time, will try to investigate this issue with MIUI.

**Comment by tribalfs** [comment]
Hi!

Please try this [new build](https://github.com/tribalfs/GalaxyMaxHzPub/releases/download/v8.0.0/galaxy_max_hz_v8.0.0.apk) if this fixed refresh rate overlay retention on MIUI.

Thank you.

**Comment by pauloup** [comment]
Hi! It's fixed! When I enable "Refresh Rate Monitor" in 8.0.0.7, the overlay is not showed anymore, only the status bar icon. Thank you!

But since version 8.0.0.6, I can't open the Refresh Rate Monitor options screen anymore. When I touch the words "Refresh Rate Monitor", only the toogle is enabled/disabled. [Like this](https://user-images.githubusercontent.com/17508852/196049396-50816628-f648-42ae-9e13-ec6496070590.mp4).

Is this the expected behavior? If not, I'm sorry, I should have reported it before. I just thought the overlay feature was disabled completely. So I can't really test if after I enable the overlay and disabled it back, it'll stay hidden.

**Comment by tribalfs** [comment]
Yes that's the expected behavior - I (temporarily) disabled overlay option on MIUI.

Is the option still not shown on the build I sent above?

**Comment by pauloup** [comment]
Yeah, the screen recording I showed was taken in 8.0.0.7, from the link you posted now. I still can't access the Refresh Rate Monitor options, only enable and disabled it.

**Comment by tribalfs** [comment]
I found the cause why still not shown.
Please try [this one](https://github.com/tribalfs/GalaxyMaxHzPub/releases/download/v8.0.0/galaxy_max_hz_v8.0.0.apk) instead.
Thank you.

**Comment by pauloup** [comment]
Not yet, I guess. Now it just don't do nothing when tapping over the words "Refresh Rate Monitor". But I can still enable and disabled it by touching the toogle.

**Comment by pauloup** [comment]
Thank you so much for taking your time to look on this, I really appreciate it

**Comment by tribalfs** [comment]
> Not yet, I guess. Now it just don't do nothing when tapping over the words "Refresh Rate Monitor". But I can still enable and disable it by touching the toogle.

Oh, my bad. Please install [this one](https://github.com/tribalfs/GalaxyMaxHzPub/releases/download/v8.0.0/galaxy_max_hz_v8.0.0.apk).

**Comment by pauloup** [comment]
Yeaaaah, now it's everything working great! Thank you so much! I enabled and disabled the overlay, and tested the "both" option, all working fine

The only issue is that I had to clean data after installing each of these APKs, otherwise the app would crash on starting.

So if there's no other way to avoid cleaning the data, I guess it may be useful to add it on the release notes, so please can find ways to backup and restore their config (like Per-app settings). I usually take screenshots to redo my settings.

**Comment by pauloup** [comment]
Again, thank you for working on this, even my device not being officially supported, therefore you're not being able to work on MIUI yourself.

That said, there other two really, really small issues that are probably related to MIUI, but I think it's worth mentioning now that we're here:

1. In the versions I tested today, if I "close" the app from Recent Apps, the app also closes in the background. The permanent notification stays in place, but the refresh rate stops changing. The overlay, if enabled, simply disappears. I have to reopen the app for it to come back working: [Screen recording](https://user-images.githubusercontent.com/17508852/196053154-11535d37-4d3b-44ec-8c05-2f58fa850b67.mp4). If the "Autostart" option is enabled in the App Info settings, the app stars a "bootloop" kinda behavior: It starts, closes, shows a toast asking to enable "Accessibility", while the Accessibility option turns on and off on its own in a loop. Pretty instable: [Screen Recording](https://photos.app.goo.gl/Paw8mYm6hWG9qhb7A). Not a big deal, I just have to keep the app open in Recent Apps. But in previous versions, closing the app from recent apps didn't stop it from working in the background. 

2. Since the first version I've tried (8.0.0), the "Standard Mode" says "FHD+ (2400x1080) @ 60 Hz / HD+ (1600x720) @ 60 Hz" at the top, but when selected, the display stays at 120hz. I don't use this option, so it doesn't affect me. But I have noticed that it doesn't follow the refresh rate chosen on the MIUI settings, and always sets 120hz: [Screenshot](https://user-images.githubusercontent.com/17508852/196052890-4f322ae1-970d-49af-82ba-25f4da900a30.jpg).

Please, Let me know if you want me to open new Issues to better talk about these, or we can keep all the MIUI related stuff in here. Or even if you just don't think they are worth the effort and time for fixing. I would totally understand that too.

**Comment by tribalfs** [comment]
1. Does it occur when refresh rate monitor overlay is turned off or refresh rate monitor completely turned off?
2.  Noted.  Will check this soon.

Thank you.

**Comment by pauloup** [comment]
1. Yes, I've disabled the Refresh Rate options and got the same issue: [Screen Recording](https://photos.app.goo.gl/3P6Zbsk9X6VgfrZ78).

Thank you!

**Comment by pauloup** [comment]
1. Yes, I've disabled the Refresh Rate options and got the same issue: [Screen Recording](https://photos.app.goo.gl/3P6Zbsk9X6VgfrZ78).

Thank you!

**Comment by tribalfs** [comment]
Please try [this build](https://github.com/tribalfs/GalaxyMaxHzPub/releases/download/v8.0.0/galaxy_max_hz_v8.0.0.apk)
Thank you.

**Comment by pauloup** [comment]
Both issues fixed! 1. Standard Mode now shows and sets 60Hz. 2. Closing the app in Recent Apps keeps the app working in background. 🥹 I'm out of words, and feel like repeating myself for thanking you everytime, but I really feel like that. Thank you for all the effort.

**Comment by tribalfs** [comment]
Great!
You are welcome.
Thank you too.
